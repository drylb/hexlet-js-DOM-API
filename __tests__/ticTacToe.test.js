/* eslint-disable no-undef */
import fs from 'fs';
import path from 'path';
import prettier from 'prettier';

import run from '../src/ticTacToe';

const getCell = (rowIndex, cellIndex) => {
  const table = document.querySelector('table');
  return table.rows.item(rowIndex).cells.item(cellIndex);
};

const options = {
  parser: 'html',
  htmlWhitespaceSensitivity: 'ignore',
  tabWidth: 4,
};

const getTree = () => prettier.format(document.body.innerHTML, options);

beforeAll(() => {
  const initHtml = fs.readFileSync(path.join('__fixtures__', 'ticTacToe.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', () => {
  expect(getTree()).toMatchSnapshot();

  getCell(2, 2).click();
  getCell(1, 1).click();
  getCell(1, 2).click();
  expect(getTree()).toMatchSnapshot();

  getCell(2, 1).click();
  expect(getTree()).toMatchSnapshot();

  getCell(2, 1).click();
  expect(getTree()).toMatchSnapshot();

  getCell(0, 0).click();
  expect(getTree()).toMatchSnapshot();

  getCell(1, 0).click();
  getCell(0, 1).click();
  expect(getTree()).toMatchSnapshot();

  getCell(2, 0).click();
  getCell(0, 2).click();
  expect(getTree()).toMatchSnapshot();
});
