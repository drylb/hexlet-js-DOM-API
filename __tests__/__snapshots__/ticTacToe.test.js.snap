// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <div class=\\"root\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <div class=\\"root\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <div class=\\"root\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <div class=\\"root\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <div class=\\"root\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
"
`;

exports[`application 6`] = `
"<div class=\\"container m-3\\">
    <div class=\\"root\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\">x</td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\"><span class=\\"invisible\\">s</span></td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
"
`;

exports[`application 7`] = `
"<div class=\\"container m-3\\">
    <div class=\\"root\\">
        <table class=\\"table-bordered\\">
            <tbody>
                <tr>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">o</td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\">x</td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
                <tr>
                    <td class=\\"py-2 px-3\\">x</td>
                    <td class=\\"py-2 px-3\\">o</td>
                    <td class=\\"py-2 px-3\\">x</td>
                </tr>
            </tbody>
        </table>
    </div>
</div>
"
`;
